src/MediaPresenter.ts

```diff
@@ -0,0 +1,157 @@
+import * as mobx from "mobx";
+
+const MAXLEN = 10;
+
+export class MediaStore {
+  file: File;
+  duration: number = 0;
+  played: number = 0;
+  constructor(newFile: File) {
+    this.file = newFile;
+    // duration of each media attached
+    this.duration = 0;
+    // played list shows how long each of the file have been played
+    this.played = 0;
+  }
+}
+
+export class MediaPresenter {
+  constructor() {
+    mobx.makeObservable(this);
+  }
+
+  @mobx.observable.deep
+  media: MediaStore[] = [];
+
+  static isImage(file: File): boolean {
+    const imageFormat = new RegExp("image/*");
+    return imageFormat.test(file.type);
+  }
+
+  static isVideo(file: File): boolean {
+    const videoFormat = new RegExp("video/*");
+    return videoFormat.test(file.type);
+  }
+
+  @mobx.action
+  addFile(newFile: File): boolean {
+    if (MediaPresenter.isImage(newFile) || MediaPresenter.isVideo(newFile)) {
+      this.media.push(new MediaStore(newFile));
+      return true;
+    }
+    return false;
+  }
+
+  @mobx.action
+  removeFile(fileIndex: number): void {
+    if (this.media.length === 0) {
+      return;
+    }
+    let index = fileIndex % this.filesLength;
+    if (index > -1) {
+      this.media.splice(index, 1);
+    }
+  }
+
+  @mobx.action
+  setDuration(fileIndex: number, duration: number): void {
+    // make sure that accessing files is % filesLength,
+    // so that it loops back to the start of the file if the video is too long
+    const modIndex = fileIndex % this.filesLength;
+    this.media[modIndex].duration = duration;
+  }
+
+  getDuration(fileIndex: number): number {
+    // make sure that accessing files is % filesLength,
+    // so that it loops back to the start of the file if the video is too long
+    const modIndex = fileIndex % this.filesLength;
+    return this.media[modIndex].duration;
+  }
+
+  // get the name of a file
+  // if index is out of bound, fileindex go to the front of the files
+  getFileName(fileIndex: number): string {
+    if (this.media.length === 0) {
+      return "";
+    }
+    return this.media[fileIndex % this.filesLength].file.name;
+  }
+
+  getFile(fileIndex: number): File {
+    return this.media[fileIndex % this.filesLength].file;
+  }
+
+  @mobx.computed
+  get filesLength(): number {
+    return this.media.length;
+  }
+
+  // shows how long each media has played
+  @mobx.action
+  incrementFilePlayed(fileIndex: number, seconds: number): void {
+    let index = fileIndex % this.filesLength;
+    this.media[index].played =
+      (this.media[index].played + seconds) % this.media[index].duration;
+  }
+
+  // reset all played media to 0
+  @mobx.action
+  resetAllPlayedFiles(): void {
+    for (let index = 0; index < this.filesLength; index++) {
+      this.media[index].played = 0;
+    }
+  }
+
+  getFilePlayed(fileIndex: number): number {
+    if (
+      fileIndex % this.filesLength === 0 ||
+      this.getDuration(fileIndex % this.filesLength) === 0
+    )
+      return 0;
```

Comment:
curly brackets please! Makes it much easier to read

Reply:
Also think about how you can pull out values into variables here to make the fn clearer